dot/core/messages.go

```diff
@@ -82,6 +104,7 @@ func (s *Service) HandleTransactionMessage(peerID peer.ID, msg *network.Transact
 	for _, tx := range txs {
 		validity, isValidTxn, err := s.validateTransaction(peerID, head, rt, tx)
 		if err != nil {
+			// Believe don't hit this if isValidTxn, but verify in review
```

Comment:
@qdm12 There are other things that can go wrong here besides wrong error type. 1) The input to the runtime call could be wrong resulting in a codec error 2) Something can go wrong (unmarshalling for instance) in the decoding of the result of the runtime call.

My intuition is that anything going wrong as part of the decoding can be treated the same. Like if its the wrong type vs not able to unmarshall or cast then it means the data is wrong. Just a difference of its an error msg from the runtime or on our end, and maybe its worth distinguishing but I'm not sure why this would affect behavior.

I feel like we should handle errors rather than panic (especially since this includes peer reporting, we need to make sure to report invalid transaction rather than just crashing)

Reply:
yeah the more I think about it the more important I think it is to handle invalid transactions (however they are invalid) and not panic

Reply:
If this is not possible from a programming perspective, you can add a panic in validateTransaction to guard against future code changes. We should not handle a programming error but we should panic instead (only programming error, not user/network/io errors obviously).

A simple example is:

```go
type MyType uint8

const (
   A MyType = iota
   B
)

func (m MyType) String() string {
  switch m {
  case A:
    return "A"
  case B:
    return "B"
  default:
    panic(fmt.Sprintf("%d not supported", m))
  } 
}
```